addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -435,6 +435,11 @@ def _l10n_es_tbai_get_importe_desglose(self, invoice):
                 desglose['DesgloseTipoOperacion']['Entrega'].update(
                     {'S1': tax_details_info_consu_vals['S1_list'],
                      'S2': tax_details_info_consu_vals['S2_list']})
+            if not desglose.get('DesgloseTipoOperacion'):
+                raise UserError(_(
+                    "In case of a foreign customer, you need to configure the tax scope on taxes:\n%s",
+                    "\n".join(invoice.line_ids.tax_ids.mapped('name'))
+                ))
```

Comment:
We should not raise in the middle of the construction of the json, but add it to the prechecks.  As this is handled by the cron, the error would risk to stay unnoticed also.  (in account_edi, we do not raise, but return it in the return dict)

Reply:
I'm sorry, you told me exactly this some time ago 😅
PR updated